anchore_engine/services/catalog/__init__.py

```diff
@@ -979,22 +985,39 @@ def handle_analyzer_queue(*args, **kwargs):
                     qobj['parent_manifest'] = parent_manifest
 
                     try:
-                        if not q_client.is_inqueue('images_to_analyze', qobj):
+                        q_record = q_client.is_inqueue('images_to_analyze', qobj)
+                        if not q_record:
                             # queue image for analysis
-                            logger.debug("queued image for analysis: " + str(imageDigest))
-                            qobj = q_client.enqueue('images_to_analyze', qobj)
-
-                            # set the appropriate analysis state for image 
-                            # image_record['analysis_status'] = taskstate.queued_state('analyze')
-                            # image_record['analysis_status'] = taskstate.working_state('analyze')
-                            # with db.session_scope() as dbsession:
-                            #    rc = db.db_catalog_image.update_record(image_record, session=dbsession)
+                            priority = False
+                            logger.debug("queued image for analysis (priority={}): {}".format(priority, str(imageDigest)))
+                            qobj = q_client.enqueue('images_to_analyze', qobj, forcefirst=priority)
 
                         else:
                             logger.debug("image already queued")
+                            # track and store the account's lowest queueId in the task queue, as well as the global highest negative space queueId across all accounts
+                            try:
+                                lowest_queueId = queue_rebalance[userId].get('lowest_queueId', None)
+                                if not lowest_queueId or q_record.get('queueId') < lowest_queueId:
+                                    queue_rebalance[userId]['lowest_queueId'] = q_record.get('queueId')
+                                if q_record.get('queueId') < 0 and q_record.get('queueId') >= highest_neg_queueId:
+                                    highest_neg_queueId = q_record.get('queueId')
+                            except Exception as err:
+                                logger.error("failed to store image current queueID - excpetion: {}".format(err))
+                                
                     except Exception as err:
                         logger.error("failed to check/queue image for analysis - exception: " + str(err))
 
+    # promote queued tasks into the analysis queue such that one image from each account is prioritized, to implement a simple 'fair share' across accounts
```

Comment:
I'd like to see this split out into its own function with some test coverage of the behavior so we can catch any regressions later.

Reply:
the logic has been adjusted to support testability, and a unit test has been added